ddtrace/contrib/sanic/patch.py

```diff
@@ -40,10 +40,18 @@ def _wrap_response_callback(span, callback):
     # based on response and finish span before returning response
 
     def update_span(response):
-        span.set_tag(http.STATUS_CODE, response.status)
-        if 500 <= response.status < 600:
-            span.error = 1
-        store_response_headers(response.headers, span, config.sanic)
+        error = 0
+        if isinstance(response, sanic.response.BaseHTTPResponse):
+            status_code = response.status
+            if 500 <= response.status < 600:
+                error = 1
+            store_response_headers(response.headers, span, config.sanic)
+        else:
+            # invalid response causes ServerError exception which must be handled
```

Comment:
Might be useful to be specific about what "invalid response" means here. Also, out of curiosity what is `response` when this happens?

Reply:
In this invalid case, the response is a `str`. One could also have a route handler that doesn't return a response, which should also be an error and should be flagged as such. I've added a test for this too.